### PR TITLE
OpenTitan: AES: Update the AES implementation

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

This updates the implemention to remove all infinite loops, use callbacks and update to the current OpenTitan bitstream addresses and registers.

### Testing Strategy

Running the Tock AES tests.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: chips/earlgrey/src/aes.rs [hunk 1]
```diff
@@ -318,32 +392,116 @@ impl<'a> hil::symmetric_encryption::AES128<'a> for Aes<'a> {
                 }
             }
         }
+
+        if self.deferred_call.get() {
+            panic!("Operation already in progress");
```
Comment (bradjc) [inline chips/earlgrey/src/aes.rs:397]: This can't just return an error?

File: chips/earlgrey/src/aes.rs [hunk 2]
```diff
@@ -318,32 +392,116 @@ impl<'a> hil::symmetric_encryption::AES128<'a> for Aes<'a> {
...
+        self.registers.ctrl.write(ctrl);
     }
 }
 
 impl kernel::hil::symmetric_encryption::AES128ECB for Aes<'_> {
     fn set_mode_aes128ecb(&self, encrypting: bool) {
-        self.configure(encrypting);
+        if !self.idle() {
+            panic!("AES is not idle");
+        }
```
Comment (bradjc) [inline chips/earlgrey/src/aes.rs:452]: This is unfortunate.

Comment (alistair23) [reply chips/earlgrey/src/aes.rs:452]: Yeah, we can't return an error so I'm not sure what else to do

Comment (alistair23) [reply chips/earlgrey/src/aes.rs:456]: I'm planning on working on capsules as well, if I end up changing the HIL I can fix this up and remove these

**Review by ppannuto — APPROVED** [review]
It would be good to fix the HIL, but in the now, I guess the panics are better than incorrect behavior.

**Review by bradjc — APPROVED** [review]
https://github.com/tock/tock/issues/2635

**Comment by alistair23** [comment]
This is ready to go

**Pull request merged**